### DISCUSSION TITLE
主题配置平台的代码不开源吗？

### DISCUSSION THREAD

**Discussion by Eviwang** [DouyinFE/semi-design#782] [Q&A]
### Which Component 出现bug的组件
  - Button

### semi-ui version
  - latest
    
### Expected result 期望的结果是什么
  - 

### Actual result 实际的结果是什么
  - 

### Steps to reproduce 复现步骤
  - 1

### Reproducible code 复现代码
```
class Demo extends React.component {

    render() {
        return (
            <Button>bug</Button>
        )
    }

}

```

### Additional information 补充说明
- 遇到这个bug的业务场景、上下文、或者你的需求场景

**Comment by pointhalo** [comment]
Semi DSM 整体是一个带服务的平台，它包括前端平台、文档数据、Figma插件以及对应的server服务，并非是无依赖的纯前端 library类型项目。这个暂时没有计划进行开源。

**Comment by chenyulun** [comment]
我们也想做一个配置平台，semi的配置平台做的挺好，想要前端这部分的代码，或者能不能请求提供一下使用的技术栈或者关键依赖，设计规范之类的？感谢

**Reply by pointhalo** [reply]
抱歉，这块没法提供源码😂
1. 首先它不是个人项目，我们要对外开放内部项目的源码的话，得遵循我们公司的开源流程。但是 DSM这个东西，它是一个整套耦合的在线服务（前后端+ Figma 插件）没法做拆分，  
同时它是跟组件的token紧密结合在一起的，并且都是跟 Semi UI 的 DOM 结构，主题方案耦合的，对于其他 ui library 并不适用。所以后续迭代主体肯定是我们自己，不太像基础 UI 一样可以依靠社区力量协助迭代。
单纯换个 origin ，脱离了我们的内部研发环境，我们还得花挺高的成本，重新调整开发部署一整套流程，这个没有实际意义，都是一些不必要的成本。现阶段我们人力有限，没法推动相关流程往下走。

2. 除了 react、semi-ui、代码编辑token部分用了 codemirror。没有其余的关键依赖了。无论交互还是逻辑，都没有现成的lib能满足，所有模块都是自己从零写的。

**Comment by chenyulun** [comment]
@pointhalo 大佬，我觉得semi的主题算法太棒了，
https://mp.weixin.qq.com/s/noHoWRuA25PgqFNcurhIUA
我和我们设计推荐了咱们的算法，感觉很强，他们最后通用使用semi design的色阶算法来自动生成，但是...
能不能把给定颜色自动基础色阶算法的源码提供一下嘛？补充到上面链接的文章里面也行，跪谢！！！
如果不想公开，可以发邮件，感谢了 308813800@qq.com
强烈对接咱们这么棒的**整套解决方案**， 点赞
